CommandHandlerTesterLib/TestLongActionPlugin.cs

```diff
@@ -0,0 +1,52 @@
+﻿using CA_DataUploaderLib;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace CommandHandlerTesterLib
+{
+    public class TestLongActionPlugin : LoopControlPlugin
+    {
+        public TestLongActionPlugin(CommandHandler cmd) : base(cmd)
+        {
+            AddCommand(nameof(LongAction), LongAction);
+            AddCommand("help", HelpMenu);
+        }
+
+        private bool HelpMenu(List<string> _)
+        { 
+            CALog.LogInfoAndConsoleLn(LogID.A, "longaction                - runs a bit longer action");
+            return true;
+        }
+
+        private bool LongAction(List<string> arg)
+        {
+            Task.Run(DoLongAction);
+            return true;
+        }
+
+        private async Task DoLongAction()
+        {
+            try
+            {
+                Console.WriteLine("Waiting for the next multiple of 10 value for IterationSensor");
+                var val = await WhenSensorValue("IterationSensor", v => v % 10 == 0, TimeSpan.FromSeconds(12));
```

Comment:
@copenhagenatomics could not come up with a way to improve on the lambda v => v % 10 == 0

only thing that comes to mind, would be potentially using expressions, so that we use the name of the lambda argument as the sensor name, like this:

`WhenSensorValue(IterationSensor => IterationSensor % 10 == 0, TimeSpan.FromSeconds(12));

But not sure that is much better.

Reply:
not better

Reply:
@copenhagenatomics then plan to keep it as is. Even with v1, plugin authors have the option of using local functions, wonder if that helps
```
bool IsMultipleOf10(double val) => val % 10 == 0;
var val = await WhenSensorValue("IterationSensor", IsMultipleOf10, Seconds(12));
```